descriptions/header.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "coding/reader.hpp"
+#include "coding/serdes_binary_header.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include <cstdint>
+
+namespace descriptions
+{
+struct HeaderV0
+{
+  template <typename Visitor>
+  void Visit(Visitor & visitor)
+  {
+    visitor(m_featuresOffset, "featuresOffset");
+    visitor(m_stringsIndexOffset, "stringsIndexOffset");
+    visitor(m_stringsOffset, "stringsOffset");
+    visitor(m_eosOffset, "eosOffset");
+  }
+
+  template <typename Sink>
+  void Serialize(Sink & sink)
+  {
+    coding::binary::HeaderSerVisitor<Sink> visitor(sink);
+    visitor(*this);
+  }
+
+  template <typename Source>
+  void Deserialize(Source & source)
+  {
+    coding::binary::HeaderDesVisitor<Source> visitor(source);
+    visitor(*this);
+  }
+
+  // Calculates the size of serialized header in bytes.
```

Comment:
Note that this is an approximation to the real size, because all these visitors do not take into account alignment of fields.